doc/book/application.md

```diff
@@ -186,21 +186,24 @@ function ($error, ServerRequestInterface $request, ResponseInterface $response,
 
 Read the section on [piping vs routing](router/piping.md) for more information.
 
-### Registering routing middleware
+### Registering routing and dispatch middleware
 
-Routing is accomplished via dedicated a dedicated middleware method, 
-`Application::routeMiddleware()`. It is an instance method, and can be
-piped/registered with other middleware platforms if desired.
+Routing is accomplished via dedicated a dedicated middleware method,
+`Application::routeMiddleware()`; similarly, dispatching of routed middleware
+has a corresponding instance middleware method, `Application::dispatchMiddleware()`.
+Each can be piped/registered with other middleware platforms if desired.
```

Comment:
If you define routes specific to the Application instance, you can pipe each of those individually to another stack. I'd likely just pipe the Application instance, however.